app/domain/operations/reports/generate_notices_report.rb

```diff
@@ -0,0 +1,87 @@
+# frozen_string_literal: true
+
+require 'dry/monads'
+require 'dry/monads/do'
+
+module Operations
+  module Reports
+    # Gets the people who has documents created in a specified date range and
+    # inserts those document information into a CSV file.
+    # Defaults to current day if no range is specified.
+    class GenerateNoticesReport
+      include Dry::Monads[:result, :do]
+
+      def call(params)
+        values            = yield validate(params)
+        people            = yield fetch_people(values)
+        report_status     = yield generate_report(people, values)
+
+        Success(report_status)
+      end
+
+      private
+
+      def validate(params)
+        start_date = params[:start_date].present? ? Date.strptime(params[:start_date].to_s, "%m/%d/%Y").beginning_of_day : TimeKeeper.date_of_record.beginning_of_day
+        end_date = params[:end_date].present? ? Date.strptime(params[:end_date].to_s, "%m/%d/%Y").end_of_day : TimeKeeper.date_of_record.end_of_day
+
+        Success(params.merge!(start_date: start_date, end_date: end_date))
+      end
+
+      def fetch_people(values)
+        people = Person.where(
+          :documents => {
+            :$elemMatch => {
+              :created_at.gte => values[:start_date],
+              :created_at.lte => values[:end_date]
+            }
+          }
+        )
+
+        Success(people)
+      end
+
+      def fetch_notice_report_file_name
+        "notices_list_#{TimeKeeper.date_of_record.strftime('%Y_%m_%d')}.csv"
+      end
+
+      def fetch_notice_report_headers
+        ['HBX ID', 'Notice Title', 'Notice Code', 'Date']
+      end
+
+      def notices_title_code_mapping
```

Comment:
- Fetching every notice from DB takes lot of time for IVL dry run scenarios.
- Instead we can add constant tags to fetch specific data for every stage of dry run, may be something like this.

```suggestion
      OE_ELIGIBILITY_NOTICES_TITLE_MAPPING = {}.freeze
      OE_ENROLLMENT_NOTICES_TITLE_MAPPING = {}.freeze
      NON_OE_ELIGIBILITY_NOTICES_TITLE_MAPPING = {}.freeze
      DR_NOTICES_TITLE_MAPPING = {}.freeze
```

Reply:
Made [separate ticket](https://www.pivotaltracker.com/story/show/188260680) to refactor this operation